src/harness/testcases/WINNF_FT_S_SIQ_testcase.py

```diff
@@ -841,6 +842,237 @@ def test_WINNF_FT_S_SIQ_11(self):
+    ppa_record2 = json.load(
+        open(os.path.join('testcases', 'testdata', 'ppa_record_1.json')))
+    
+    pal_low_frequency1 = 3570000000
+    pal_high_frequency1 = 3580000000
+    pal_user_id1 = "pal_user_1"
+    pal_low_frequency2 = 3600000000
+    pal_high_frequency2 = 3610000000
+    pal_user_id2 = "pal_user_2"
+
+    # Make PAL and PPA record consistent
+    ppa_record1, pal_records1 = makePpaAndPalRecordsConsistent(
+        ppa_record1, [pal_record_0], pal_low_frequency1, pal_high_frequency1,
+        pal_user_id1)
+    ppa_record2, pal_records2 = makePpaAndPalRecordsConsistent(
+        ppa_record2, [pal_record_1], pal_low_frequency2, pal_high_frequency2,
+        pal_user_id2) 
+    
+    ppa_cluster_list_1 = [0]
+    ppa_cluster_list_2 = []
+    # Move device_b inside the PPA1 zone
+    device_b['installationParam']['latitude'] = 38.82767
+    device_b['installationParam']['longitude'] = -97.20497
+
+    # Creating conditionals for Cat B devices
+    self.assertEqual(device_b['cbsdCategory'], 'B')
+    conditionalParameters = {
+        'cbsdCategory': device_b['cbsdCategory'],
+        'fccId': device_b['fccId'],
+        'cbsdSerialNumber': device_b['cbsdSerialNumber'],
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam'],
+        'measCapability': device_b['measCapability']
+    }
+    del device_b['cbsdCategory']
+    del device_b['airInterface']
+    del device_b['installationParam']
+
+    # N2 spectrum inquiry requests
+    # 1. Spectrum Inquiry: 3550-3580 .
+    spectrum_inquiry_1 = json.load(
+        open(os.path.join('testcases', 'testdata', 'spectrum_inquiry_0.json')))
+    spectrum_inquiry_1['inquiredSpectrum'] = [{
+        'lowFrequency': 3550000000,
+        'highFrequency': 3580000000
+    }]
+
+    # 2. Spectrum Inquiry: Invalid Value of low and high frequency and invalid id for cbsdId.
+    spectrum_inquiry_2 = json.load(
+        open(os.path.join('testcases', 'testdata', 'spectrum_inquiry_0.json')))
+    spectrum_inquiry_2['inquiredSpectrum'] = [{
+        'lowFrequency': 3610000000,
+        'highFrequency': 3590000000
+    }]
+    spectrum_inquiry_2['cbsdId'] = "INVALID_CBSD_ID_12345"
+
+    # 3. Spectrum Inquiry: Missing parameter low frequency and missing cbsdId
+    spectrum_inquiry_3 = json.load(
+        open(os.path.join('testcases', 'testdata', 'spectrum_inquiry_0.json')))
+    spectrum_inquiry_3['inquiredSpectrum'] = [{
+        'highFrequency': 3700000000
+    }]
+    spectrum_inquiry_3['cbsdId'] = "REMOVE"
+
+    # Create the actual config.
+    devices = [device_b, device_c, device_e]
+    conditionals = {
+        'registrationData': [conditionalParameters]
+    }
+
+    config = {
+        'registrationRequestsN2': devices,
+        'conditionalRegistrationData' : conditionals,
+        'expectedResponseCodes': [(0,), (103,), (102,)],  
+        'palRecordsN3':[pal_records1,pal_records2],
```

Comment:
This should be just a list, not a list of lists. Having it as a list of lists leads to another bug later, where you only take the first value of each sub-list (which causes no problems with this default config but would cause problems with a more generic config).

Reply:
Is it okay to have a list of lists, but iterate the list of lists in later code? The palRecords list should match the ppaRecords list, so each palRecords list is associated with each ppaRecord.

Reply:
Actually, I don't think you can assume that they are matched in that way. One PAL record can have two associated PPAs, which would break this scheme. I believe the best approach is to have a single list of PAL records which is not linked directly to the PPAs -- we don't use the linkage anyway, and if we were using it, it would be best recovered by using the PAL IDs inside the PPA record.

Reply:
Thank you. Agree with these good points, and made the fix to have one PAL records list without links to PPAs.